mne/tests/test_annotations.py

```diff
@@ -635,4 +636,132 @@ def _constant_id(*args, **kwargs):
+    content = ("onset,duration,description\n"
+               "2002-12-03 19:01:11.720100,1.0,AA\n"
+               "2002-12-03 19:01:20.720100,2.425,BB")
+
+    fname = tmpdir_factory.mktemp('data').join('annotations.csv')
+    fname.write(content)
+    return fname
+
+
+@requires_pandas
+def test_io_annotation_csv(dummy_annotation_csv_file, tmpdir_factory):
+    annot = read_annotations(str(dummy_annotation_csv_file))
+    assert annot.orig_time == 1038942071.7201
+    assert_array_equal(annot.onset, np.array([0., 9.], dtype=np.float32))
+    assert_array_almost_equal(annot.duration, [1., 2.425])
+    assert_array_equal(annot.description, ['AA', 'BB'])
+
+    # Now test writing
+    fname = str(tmpdir_factory.mktemp('data').join('annotations.csv'))
+    annot.save(fname)
+    annot2 = read_annotations(fname)
+    _assert_annotations_equal(annot, annot2)
+
+    # Now without an orig_time
+    annot.orig_time = None
+    annot.save(fname)
+    annot2 = read_annotations(fname)
+    _assert_annotations_equal(annot, annot2)
+
+
+# Test for IO with .txt files
+
+@pytest.fixture(scope='session')
+def dummy_annotation_txt_file(tmpdir_factory):
+    content = ("3.14, 42, AA \n"
+               "6.28, 48, BB")
+
+    fname = tmpdir_factory.mktemp('data').join('annotations.txt')
+    fname.write(content)
+    return fname
+
+
+def test_io_annotation_txt(dummy_annotation_txt_file, tmpdir_factory):
+    annot = read_annotations(str(dummy_annotation_txt_file))
+    assert annot.orig_time is None
+    assert_array_equal(annot.onset, np.array([3.14, 6.28], dtype=np.float32))
+    assert_array_equal(annot.duration, np.array([42., 48], dtype=np.float32))
+    assert_array_equal(annot.description, ['AA', 'BB'])
+
+    # Now test writing
+    fname = str(tmpdir_factory.mktemp('data').join('annotations.txt'))
+    annot.save(fname)
+    annot2 = read_annotations(fname)
+    _assert_annotations_equal(annot, annot2)
+
+    # Now with an orig_time
+    annot.orig_time = 1038942071.7201
+    annot.save(fname)
+    annot2 = read_annotations(fname)
+    _assert_annotations_equal(annot, annot2)
+
+
+@pytest.fixture(scope='session')
+def dummy_annotation_txt_header(tmpdir_factory):
+    content = ("# A something \n"
+               "# orig_time : 42\n"
+               "# orig_time : 2002-12-03 19:01:11.720100\n"
+               "# orig_time : 42\n"
+               "# C\n"
+               "Done")
+    fname = tmpdir_factory.mktemp('data').join('xx.txt')
+    fname.write(content)
+    return fname
+
+
+@pytest.mark.parametrize('meas_date, out', [
+    pytest.param('toto', 0, id='invalid string'),
+    pytest.param(None, 0, id='None'),
+    pytest.param(42, 42.0, id='Scalar'),
+    pytest.param(3.14, 3.14, id='Float'),
+    pytest.param((3, 140000), 3.14, id='Scalar touple'),
+    pytest.param('2002-12-03 19:01:11.720100', 1038942071.7201,
+                 id='valid iso8601 string'),
+    pytest.param('2002-12-03T19:01:11.720100', 0,
```

Comment:
Why is this an invalid string? I thought it'd be good to also test date strings with the funny `T` character separating date from time (which should be valid according to the regex).

Reply:
I've only implemented ' '. Thats why I made this test, to show that does not work. It is a valid ISO, but we don't parse it.